src/main.c

```diff
@@ -1,72 +1,183 @@
-			printf("%d ", vec[i]);
+int cliCmd(int argc, char const * argv[], FILE * files[]) {
+	int valueToReturn = EXIT_SUCCESS;
+	files[INPUT_POS] = NULL;
+	files[OUTPUT_POS] = NULL;
+
+	if (argc >  5) {
+		fprintf(stderr, "Too many parameters -h for help\n");
+		//write(2, "Too many parameters -h for help\n", 0);
+		return FINISH;
+	}
+	int skip_row = 0;
+	for(size_t i = 1; i < argc; i++){
+		if (skip_row) {
+			skip_row = 0;
+			continue;
 		}
+		if (!(strcmp(argv[i], "-h") && strcmp(argv[i], "--help"))){
+			fprintf(stdout, "%s", MSG_HELP);
+			valueToReturn = FINISH;
+			break;
+		}
+		else if (!(strcmp(argv[i], "-V") && strcmp(argv[i], "--version"))){
+			fprintf(stdout, "%.2f", VERSION);
+			valueToReturn = FINISH;
+			break;
+		}
+		else if (!(strcmp(argv[i], "-i") && strcmp(argv[i], "--input"))){
+			if(strcmp("-",argv[i+1])){
+				files[INPUT_POS] = fopen(argv[i+1],"r");
+				if(!files[INPUT_POS]){
+					fprintf(stderr, "%s", MSG_ERROR_OPEN_INPUT);
+					valueToReturn = EXIT_FAILURE;
+					break;
+				}
+			}else{
+				files[INPUT_POS] = stdin;
+			}
+			skip_row = 1;
+		}
+		else if (!(strcmp(argv[i], "-o") && strcmp(argv[i], "--output"))){
+			if(strcmp("-",argv[i+1])) {
+				files[OUTPUT_POS] = fopen(argv[i+1],"w");
+				if(!files[OUTPUT_POS]){
+					fprintf(stderr, "%s", MSG_ERROR_OPEN_OUTPUT);
+					valueToReturn = EXIT_FAILURE;
+					break;
+				}
+			}else {
+				files[OUTPUT_POS] = stdout;
+			}
+			skip_row = 1;
+		}
+		else {
+			fprintf(stderr, "%s", MSG_COMMAND_UNKNOWN);
+			valueToReturn = EXIT_FAILURE;
+			break;
+		}
+	}
 
-		printf("\n");
-		free(vec);
-	} 
-	
-	
-	// Parsear CLI nacho
-	// Apertura de archivo
-	// Inicio de while != feof(archivo)
-	// Intentar parsear linea
-	// aplicar mergesort
-	// printer en file output
-	// repetir ciclo
-	// cerrar archivos
-	return 0;
+	return valueToReturn;
 }
 
  int parser(char * line, int ** result, size_t * size_result){
 	if(!line) 			return EXIT_FAILURE; //No puede ser null
 	if(line[0] == '\n') return EMPTY_LINE;   //Catchear arriba para no aplicar mergesort
-	
+
 	size_t n_sep = 0;
 	for (size_t i=0; line[i]; i++)
 		if(line[i] == ' ') n_sep++;
-
 	int * vector = (int *) calloc(n_sep + 2, sizeof(int));
-	if (!vector) return EXIT_FAILURE;
-
+	int maxValue = n_sep + 2;
+	if (!vector){
+		fprintf(stderr, "%s", MSG_ERROR_MALLOC);
+		return EXIT_FAILURE;
+	}
 	size_t size = 0;
 	long tmp;
-	
+
 	for(char ** ptr2str = &line; **ptr2str != '\0'; (*ptr2str)++){
 		if(**ptr2str == ' ' )  continue;
 		if(**ptr2str == '\n' ) break;
-		
+
 		tmp = strtol(*ptr2str, ptr2str, DECIMAL); //Si el resultado no se puede representar devuelve cero, la funcion adelanta ptr2str hasta el proximo caracter numerico luego del num leido
-		if(tmp > MAX_INT_32 || tmp < MIN_INT_32) 
+		if(tmp > MAX_INT_32 || tmp < MIN_INT_32)
 			tmp = 0;
 
 		vector[size++] = (int) tmp;
+		if (!(size < maxValue)) {
+			maxValue *= 2;
+		}
+		vector = (int*) realloc(vector, sizeof(int) * maxValue);
```

Comment:
No entiendo porque este realloc. Si se llegara a dar el caso de que el size sea mayor (by the way size > maxvalue queda mejor en el if jaja) y se tuviera que hacer un realloc entonces habria que recorrer toda la linea de nuevo. Igualmente al contar la cantidad de separadores que hay en esa linea nos aseguramos de tener espacio suficiente para todos los elementos. Ahora que lo estoy pensando en un caso que tengamos " 1 2.     4 5" estariamos reservando memoria de sobra, capaz sea mejor contar cada vez que hay un separador entre dos cosas distintas de un separador. Me parece que no hay restricciones de uso de memoria en el TP, pero seria una buena optimizacion

Reply:
Es una optimizacion innecesaria que tal vez requiera mas trabajo que no se si es necesario para este tp. 
https://en.wikipedia.org/wiki/Gold_plating_(project_management)